internal/core/operations/update_scheduler/update_scheduler_executor.go

```diff
@@ -37,8 +37,9 @@ import (
 )
 
 type UpdateSchedulerExecutor struct {
-	roomManager      *room_manager.RoomManager
-	schedulerManager interfaces.SchedulerManager
+	roomManager        *room_manager.RoomManager
+	schedulerManager   interfaces.SchedulerManager
+	roomsBeingReplaced sync.Map
```

Comment:
In my first implementation, i was using a build-in map with a Mutex to manage concurrency, but then I discovered this awesome map that simplifies the code and can be used for what we are doing here.